packages/core/src/index.ts

```diff
@@ -500,10 +502,44 @@ class Hopr extends EventEmitter {
     } else {
       log(`No multiaddrs has been registered.`)
     }
+
+    // Public relay nodes always run the DHT in server-mode
+    // so only switch modes if node was started without the `--announce` flag.
+    if (!this.options.announce) {
+      await this.maybeEnableDhtServerMode()
+    }
+
     await this.maybeLogProfilingToGCloud()
+
+    console.log(this.libp2pComponents.getRegistrar().getProtocols())
+
     this.heartbeat.recalculateNetworkHealth()
   }
 
+  /**
+   * Checks if we are announcing public addresses to the DHT.
+   * If so, switch DHT to `server`-mode such that the node will
+   * reply to DHT queries of other nodes
+   */
+  private async maybeEnableDhtServerMode() {
+    let dht: Components['dht']
+    try {
+      dht = this.libp2pComponents.getDHT()
+    } catch (err) {
+      error(`Cannot switch DHT to server mode:`, err)
+      return
+    }
+
+    let announcedAddresses = this.libp2pComponents.getTransportManager().getAddrs()
+
+    for (const addr of announcedAddresses) {
+      if ([AddressClass.Public, AddressClass.Public6].includes(maToClass(addr))) {
```

Comment:
This isn't deterministic. As of now, even nodes behind a NAT will add an entry to the DHT which includes their public IP. Thus, these would be set to server mode as well.

Reply:
Depends on the type of NAT, in case of permissive NATs, it *won't* publish any external address. E.g. run `hopr` on a mobile hotspot or in the office